core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -524,9 +524,15 @@ class GroupMetadataManager(brokerId: Int,
           val memRecords = fetchDataInfo.records match {
             case records: MemoryRecords => records
             case fileRecords: FileRecords =>
-              buffer.clear()
-              fileRecords.readInto(buffer, 0)
-              MemoryRecords.readableRecords(buffer)
+              val recordsBuffer = if (fileRecords.sizeInBytes > buffer.capacity) {
+                debug(s"Loaded offsets and group metadata from $topicPartition with buffer larger than configured offsets.load.buffer.size")
```

Comment:
It would be better if we change the log level to WARN and add the value of fileRecords.sizeInBytes to the log message.